ducktape/command_line/main.py

```diff
@@ -171,6 +171,7 @@ def main():
     # Run the tests
     runner = SerialTestRunner(session_context, tests)
     test_results = runner.run_all_tests()
+    session_context.close_logger()
```

Comment:
Yeah I hear your point. It was a little unclear what `close` ought to mean wrt test_context/session_context

test teardown involves:
- try to stop services associated with test
- collect logs associated with test
- clean processes/logs/scratch space
(Maybe the following reasonably belong in "close"?)
- release service nodes back to cluster
- delete reference to services
- close log handles